packages/suite-desktop-core/src/modules/trezor-connect.ts

```diff
@@ -99,10 +105,23 @@ export const initBackground: ModuleInitBackground = ({ mainThreadEmitter, store
     return { onLoad, onQuit };
 };
 
-export const init: ModuleInit = () => {
+export const init: ModuleInit = ({ mainThreadEmitter }) => {
     const onLoad = () => {
         // reset previous instance, possible left over after renderer refresh (F5)
         TrezorConnect.dispose();
+        mainThreadEmitter.on('module/firmware/list', (event: LocalFirmwares) => {
```

Comment:
The `onLoad` is called on every desktop app reload (e.g. on Ctrl+R), so I'd rather unsubscribe previous listener from `mainThreadEmitter`, or even better, subscribe it right above `onLoad` in `init` which is executed only once?

Reply:
Ok, that makes sense move it to right before `onLoad`.

I also moved the subscription `TrezorConnect.on(DEVICE_EVENT, event =>` to `onLoad` in `init` right after `TrezorConnect.dispose();` it is not part of the scope of this PR but it was not working where it was since `dispose()` unsubscribe all the events.

https://github.com/trezor/trezor-suite/pull/18564/files#diff-6d770fd4f2933ede123857d15a676c2472d44d9bb5d820a2082803913d33fc75R102

Reply:
Hmm, that may be a bug. But before it was in `initBackground`'s `onLoad`, now it's in `init`'s `onLoad`. I'm not familiar with this relatively new architecture, but it could mean that some background-only functionality will stop working (tray module?). WDYT @martykan?

Reply:
Ok, I reverted that change here https://github.com/trezor/trezor-suite/pull/18564/commits/b64648aeb29541950e533f56cdf338d657e89d61 and created new PR only with that change so we can talk about it there https://github.com/trezor/trezor-suite/pull/18716 since it is not the scope of this PR.